src/types.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  Context,
+  Fluent,
+  FluentBundleOptions,
+  FluentOptions,
+  LocaleId,
+  TranslationContext,
+} from "./deps.ts";
+
+export type LocaleNegotiator<C extends Context = Context> = (ctx: C) =>
+  | LocaleId
+  | undefined
+  | PromiseLike<LocaleId | undefined>;
+
+export type TranslateFunction = (
+  key: string,
+  context?: TranslationContext,
+) => string;
+
+export interface I18nContextFlavor {
+  /** I18n context namespace object */
+  i18n: {
+    /** Fluent instance used internally. */
+    fluent: Fluent;
+    /** Returns the current locale. */
+    getLocale(): Promise<string>;
+    /**
+     * Equivalent for manually setting the locale in session and calling
+     * `renegotiateLocale()`. If the `useSession` in the i18n configuration is
+     * set to true, sets the locale in session. Otherwise throws an error.
```

Comment:
```suggestion
     * Equivalent for manually setting the locale in session and calling `renegotiateLocale()`.
     * If the `useSession` in the i18n configuration is set to true, sets the locale in session.
     * Otherwise throws an error.
```